src/sources/socket/mod.rs

```diff
@@ -112,26 +112,22 @@ impl SourceConfig for SocketConfig {
     async fn build(&self, cx: SourceContext) -> crate::Result<super::Source> {
         match self.mode.clone() {
             Mode::Tcp(config) => {
-                let (framing, decoding) = match (config.framing(), config.max_length()) {
-                    (Some(_), Some(_)) => {
-                        return Err("Using `max_length` is deprecated and does not have any effect when framing is provided. Configure `max_length` on the framing config instead.".into());
-                    }
-                    (Some(framing), None) => {
-                        let decoding = config.decoding().clone();
-                        let framing = framing.clone();
-                        (framing, decoding)
+                let decoding = config.decoding().clone();
+                // TODO: in v0.30.0 , remove the `max_length` setting from
+                // the UnixConfig, and all of the below mess and replace
+                // it with the configured framing /
+                // decoding.default_stream_framing().
+                let framing = match (config.framing().clone(), config.max_length()) {
+                    (Some(framing), Some(_)) => {
+                        warn!(message = "The `max_length` setting is deprecated and will be removed in an upcoming release. Since a `framing` setting was provided, the `max_length` setting has no effect.");
```

Comment:
Per the deprecation policy this should be prefixed with `DEPRECATION: `.

Reply:
This could maybe be enforced by a link that scans for warn! + "deprecat" 💭

Reply:
We could also create a `deprecated!` macro that prepends the `DEPRECATION`.

Reply:
I'd be fine with that macro/fn - but then we're at the point that you've got to remember to use `deprecated!` instead of a `warn!`, though I think that's still an improvement as the original deprecation log was at ERROR...